fetch_response.go

```diff
@@ -150,7 +150,7 @@ func (b *FetchResponseBlock) decode(pd packetDecoder, version int16) (err error)
 			return err
 		}
 
-		b.LastRecordsBatchOffset, err = records.recordsOffset()
+		b.RecordsNextOffset, err = records.nextOffset()
```

Comment:
I guess, since we’re more caching a value here rather than this `RecordsNextOffset` existing properly in the serialized communication, that there’s no reason not to precalculate it… also, since it gets stored as a pointer, incrementing it would require testing for possible `nil` anyways…